Subject: Call populate_rootfs() prior to acpi_early_init()
From: Hannes Reinecke <hare@suse.de>

populate_rootfs should be called prior to acpi_early_init()
if we ever want to be able to use a custom DSDT from initramfs.
Having done this, rootfs is available during acpi_early_init()
and the ACPI code can just call sys_open() to load a custom
DSDT.


--- linux-2.6.10/init/main.c.orig	2005-01-27 16:25:44.702565168 +0100
+++ linux-2.6.10/init/main.c	2005-01-27 16:26:37.003614200 +0100
@@ -543,8 +543,6 @@ asmlinkage void __init start_kernel(void
 #endif
 	check_bugs();
 
-	acpi_early_init(); /* before LAPIC and SMP init */
-
 	/* Do the rest non-__init'ed, we're now alive */
 	rest_init();
 }
@@ -670,6 +668,14 @@ static int init(void * unused)
 	 */
 	child_reaper = current;
 
+	/*
+	 * Do this before initcalls, because some drivers want to access
+	 * firmware files.
+	 */
+	populate_rootfs();
+
+	acpi_early_init(); /* before LAPIC and SMP init */
+
 	/* Sets up cpus_possible() */
 	smp_prepare_cpus(max_cpus);
 
@@ -679,12 +685,6 @@ static int init(void * unused)
 	smp_init();
 	sched_init_smp();
 
-	/*
-	 * Do this before initcalls, because some drivers want to access
-	 * firmware files.
-	 */
-	populate_rootfs();
-
 	do_basic_setup();
 
 	/*
